nsqd/stats.go

```diff
@@ -52,7 +53,7 @@ func NewChannelStats(c *Channel, clients []ClientStats) ChannelStats {
 		BackendDepth:  c.backend.Depth(),
 		InFlightCount: len(c.inFlightMessages),
 		DeferredCount: len(c.deferredMessages),
-		MessageCount:  c.messageCount,
+		MessageCount:  atomic.LoadUint64(&c.messageCount),
```

Comment:
Yeah, the race detector yelled at me.

Tangentially related, I think the test.sh script may not be actually failing if race conditions are found. It looks like `go test -race` still returns an exit code of `0` if the tests pass, but a race is found. I just noticed this race is indeed happening on master.